test/integration/authz_v1_test.go

```diff
@@ -0,0 +1,388 @@
+			signedDomain.Domain.Policies.KeyId = "col-env-1.2"
+			signedDomain.KeyId = "col-env-1.2"
+		},
+	}
+
+	updatedR := fixtures.GetExpectedResources(o)
+	rolloutAndValidate(t, updatedR, update)
+	cleanup(t, r)
+}
+
+// 2.4 Update role name and expect the old SR/SRB to be deleted
+func TestUpdateRoleName(t *testing.T) {
+	r := fixtures.GetExpectedResources(&fixtures.OverrideResources{})
+	rolloutAndValidate(t, r, create)
+
+	o := &fixtures.OverrideResources{
+		ModifyAD: func(signedDomain *zms.SignedDomain) {
+			signedDomain.Domain.Policies.Contents.Policies[0].Assertions[0].Role = "athenz.domain:role.client-reader-role"
+			signedDomain.Domain.Roles[0].Name = "athenz.domain:role.client-reader-role"
+		},
+	}
+
+	updatedR := fixtures.GetExpectedResources(o)
+	rolloutAndValidate(t, updatedR, update)
+	for _, config := range r.ModelConfigs {
+		c := framework.Global.IstioClientset.Get(config.Type, config.Name, config.Namespace)
+		assert.Nil(t, c, "istio custom resource get should return nil")
+	}
+	cleanup(t, updatedR)
+}
+
+// 2.5 Test updates with multiple namespaces / AD
+func TestMultipleAthenzDomain(t *testing.T) {
+	rOne := fixtures.GetExpectedResources(&fixtures.OverrideResources{})
+	rolloutAndValidate(t, rOne, create)
+
+	o := &fixtures.OverrideResources{
+		ModifyAD: func(signedDomain *zms.SignedDomain) {
+			signedDomain.Domain.Name = "athenz.domain.one"
+			signedDomain.Domain.Policies.Contents.Policies[0].Assertions[0].Role = "athenz.domain.one:role.client-reader-role"
+			signedDomain.Domain.Policies.Contents.Policies[0].Assertions[0].Resource = "athenz.domain.one:svc.my-service-name"
+			signedDomain.Domain.Roles[0].Name = "athenz.domain.one:role.client-reader-role"
+		},
+	}
+
+	rTwo := fixtures.GetExpectedResources(o)
+	rolloutAndValidate(t, rTwo, create)
+
+	o = &fixtures.OverrideResources{
+		ModifyAD: func(signedDomain *zms.SignedDomain) {
+			signedDomain.Domain.Name = "athenz.domain.two"
+			signedDomain.Domain.Policies.Contents.Policies[0].Assertions[0].Role = "athenz.domain.two:role.client-reader-role"
+			signedDomain.Domain.Policies.Contents.Policies[0].Assertions[0].Resource = "athenz.domain.two:svc.my-service-name"
+			signedDomain.Domain.Roles[0].Name = "athenz.domain.two:role.client-reader-role"
+		},
+	}
+
+	rThree := fixtures.GetExpectedResources(o)
+	rolloutAndValidate(t, rThree, create)
+
+	cleanup(t, rOne)
+	cleanup(t, rTwo)
+	cleanup(t, rThree)
+}
+
+// 3.0 Delete athenz domain
+// TODO: currently the controller does not delete the service role / binding
+// due to checking for the existence of the athenz domain. This test assumes
+// these still exist. Update this test once the controller core logic changes.
+func TestAthenzDomainDelete(t *testing.T) {
+	r := fixtures.GetExpectedResources(&fixtures.OverrideResources{})
+	rolloutAndValidate(t, r, create)
+
+	err := framework.Global.AthenzDomainClientset.AthenzV1().AthenzDomains().Delete(r.AD.Name, &v1.DeleteOptions{})
```

Comment:
would be nice if we can make `rolloutAndValidate` handle `delete` similar to `create` & `update` as well

Reply:
That's a great point, updated it.